File: ViewModelExample/ViewModelExampleJava/app/src/main/java/com/tinlone/viewmodelexamplejava/MyViewModelFactory.java
package com.tinlone.viewmodelexamplejava;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

public class MyViewModelFactory implements ViewModelProvider.Factory {

   private String defaultConfig = "";

    public MyViewModelFactory(String defaultConfig) {
        this.defaultConfig = defaultConfig;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new HomeDataViewModel(defaultConfig);
    }
}

File: ViewModelExample/ViewModelExampleKotlin/app/src/main/java/com/tinlone/viewmodelexamplekotlin/MyViewModelFactory.kt
package com.tinlone.viewmodelexamplekotlin

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

class MyViewModelFactory(defaultConfig: String) : ViewModelProvider.Factory {
    private var defaultConfig = ""
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return HomeDataViewModel(defaultConfig) as T
    }

    init {
        this.defaultConfig = defaultConfig
    }
}